lib/encoding.js

```diff
@@ -6,12 +6,13 @@
  * @fileoverview Global |this| required for resolving indexes in node.
  * @suppress {globalThis}
  */
-if (typeof module !== "undefined" && module.exports &&
-    !this["encoding-indexes"]) {
-    require("./encoding-indexes.js");
-}
-
-(function(global) {
+(function(global, factory) {
```

Comment:
Rather than introducing this new IIIFE and factory, why not move the `if (typeof module ...` block inside the original IIFE and pass `this || {}` in, as you did in the other file?